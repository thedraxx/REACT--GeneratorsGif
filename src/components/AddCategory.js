import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setinputValue] = useState("Search a GIF!");

    const handeInputChange = (e) =>{
        setinputValue(e.target.value)

        console.log('handeInputChange llamado')
    }

    const handleSubmit = (e) => {
        
        e.preventDefault();
  
        if( inputValue.trim().length > 2){
            setCategories(categories => [ inputValue,...categories,]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit = {handleSubmit}>
            <p>{inputValue}</p>
                <input
                    type="text" 
                    value={inputValue}
                    onChange={ handeInputChange}
                />

        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
  };
