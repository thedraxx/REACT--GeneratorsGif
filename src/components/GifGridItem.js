import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({tittle,url}) => {

    return (
        <div className = "card animate__animated animate__fadeIn">
           <img src = {url} alt={tittle} />
        </div>
    )
}

GifGridItem.propTypes = {
    tittle: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}



