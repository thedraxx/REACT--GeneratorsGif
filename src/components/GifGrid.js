import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import '../index.css';

export const  GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <div className='Contenedor-titulo'>
          
       
            {loading && <p className= "card animate__animated animate__flash">Loading</p> }
       
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem 
                                key={img.id}
                                {...img}
                            />
                    ))
                }
            </div>
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
