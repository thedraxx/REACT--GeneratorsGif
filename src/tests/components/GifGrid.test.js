import React from 'react';
import {GifGrid} from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';

jest.mock('../../hooks/useFetchGifs');
 
describe('Tests for GifGrid component', () => {
    const category = 'Hola Mundo';

    // const wrapper = shallow(<GifGrid category={category}/>);
 
    test('debe mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot(); 
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id:'abc',
            url:'https://localhost/dasd/cualquiercosa',
            tittle:'Cualquier cosa'
        }]; 

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe( gifs.length   )

    })
    

   
    



})