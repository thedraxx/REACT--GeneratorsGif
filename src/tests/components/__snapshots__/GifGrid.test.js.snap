// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifGrid component debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    HOLA MUNDO
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
